controller/api/destination/endpoint_translator.go

```diff
@@ -306,45 +313,52 @@ func toAddr(address watcher.Address) (*net.TcpAddress, error) {
 	}, nil
 }
 
-func toWeightedAddr(address watcher.Address, enableH2Upgrade bool, identityTrustDomain string, controllerNS string, log *logging.Entry) (*pb.WeightedAddr, error) {
-	controllerNSLabel := address.Pod.Labels[k8s.ControllerNSLabel]
-	sa, ns := k8s.GetServiceAccountAndNS(address.Pod)
-	labels := k8s.GetPodLabels(address.OwnerKind, address.OwnerName, address.Pod)
+func toWeightedAddr(address watcher.Address, opaquePorts map[uint32]struct{}, enableH2Upgrade bool, identityTrustDomain string, controllerNS string, log *logging.Entry) (*pb.WeightedAddr, error) {
+	controllerNSLabel := address.Pod.Labels[pkgk8s.ControllerNSLabel]
+	sa, ns := pkgk8s.GetServiceAccountAndNS(address.Pod)
+	labels := pkgk8s.GetPodLabels(address.OwnerKind, address.OwnerName, address.Pod)
 
 	// If the pod is controlled by any Linkerd control plane, then it can be
 	// hinted that this destination knows H2 (and handles our orig-proto
-	// translation) and supports receiving opaque traffic.
+	// translation)
 	var hint *pb.ProtocolHint
 	if enableH2Upgrade && controllerNSLabel != "" {
-		// Get the inbound port from the proxy container's environment
-		// variable so that it can be set in the protocol hint.
-		var inboundPort uint32
-	loop:
-		for _, containerSpec := range address.Pod.Spec.Containers {
-			if containerSpec.Name != k8s.ProxyContainerName {
-				continue
+		hint = &pb.ProtocolHint{
+			Protocol: &pb.ProtocolHint_H2_{
+				H2: &pb.ProtocolHint_H2{},
+			},
+		}
+		opaquePortMatches := false
+		for opaquePort := range opaquePorts {
```

Comment:
Since `opaquePorts` is a map, why iterate through all ports. Can we not do a simple lookup like `opaquePorts[address.Port]`?

Reply:
Yep we can do that. I removed `opaquePortMatches` completely from this as well.